### PR TITLE
Migrate obs-infraobs-integrations to package-spec v3 #5

### PR DESCRIPTION
**Pull request by shmsr**
Cherry-picked changes for 2 packages to this branch so that https://github.com/elastic/integrations/pull/8171 gets unblocked:

- [x] mongodb
- [x] influxdb

## Proposed commit message

Migrate the following package to use package-spec v3:

- [x] mongodb
- [x] influxdb

Here influxdb ci is failing for some unknown reason i.e., it failing with an error where it shouldn't. And mongodb needs to be tested to check if no data loss is there (tsdb).

## Checklist

- [ ] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [ ] I have verified that all data streams collect metrics or logs.
- [ ] I have added an entry to my package's `changelog.yml` file.
- [ ] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

### CODE REVIEW

File: packages/influxdb/data_stream/advstatus/manifest.yml [hunk 1]
```diff
@@ -1,5 +1,6 @@
 title: "InfluxDB database advanced status metrics"
 type: metrics
+release: beta
```
Comment (tommyers-elastic) [inline packages/influxdb/data_stream/advstatus/manifest.yml:3]: same question as elsewhere about the potential for a change in this data stream from ga -> beta with this change.

**Review by tommyers-elastic — APPROVED** [review]
LGTM - only question is about addition of `beta` release tags to data streams. thanks!

**Comment by shmsr** [comment]
@tommyers-elastic Do you know someone who can help with the CI issue we're facing with influxdb?

Ref: https://github.com/elastic/integrations/pull/8171#issuecomment-1766926003

**Comment by shmsr** [comment]
https://github.com/elastic/integrations/issues/8028#issuecomment-1770420786

We're good with mongodb tsdb testing ^^

File: packages/mongodb/docs/README.md [hunk 2]
```diff
@@ -407,7 +407,7 @@ The fields reported are:
 | mongodb.collstats.total.time.us | Total waiting time for locks in microseconds. | long | counter |
 | mongodb.collstats.update.count | Number of document update events. | long | counter |
 | mongodb.collstats.update.time.us | Time updating documents in microseconds. | long | counter |
-| service.address | Address of the machine where the service is running. | keyword |  |
+| service.address | Address where data about this service was collected from. This should be a URI, network address (ipv4:port or [ipv6]:port) or a resource path (sockets). | keyword |  |
```
[the remark this answers is not in the document]
Comment (shmsr) [reply packages/mongodb/docs/README.md:410]: > Address of the machine where the service is running.

This one is good enough but what I like about the description from ECS is that it is explicit about what's accepted as an address — URI, network address, resource path (UNIX sockets?). The simpler definition does not clearly convey that. What do you think?

File: packages/influxdb/data_stream/advstatus/manifest.yml (hunk 1 above)
Comment (shmsr) [reply packages/influxdb/data_stream/advstatus/manifest.yml:3]: https://github.com/elastic/integrations/issues/8028#issuecomment-1769962013

as per this removing this.

**Comment by shmsr** [comment]
Upgraded the stack version for influxdb to ^8.9.0 to fix the constant CI failures:

> Error: error running package system tests: could not complete test run: failed to install package: can't install the package: could not zip-install package; API status code = 500; response body = {"statusCode":500,"error":"Internal Server Error","message":"illegal_argument_exception\n\tRoot causes:\n\t\tillegal_argument_exception: Limit of total dimension fields [21] has been exceeded"}

**Comment by shmsr** [comment]
@tommyers-elastic Should I merge this PR?

**Comment by tommyers-elastic** [comment]
i'm happy to upgrade the stack version to make the CI more reliable. @lalit-satapathy how about you?

**Comment by lalit-satapathy** [comment]
> i'm happy to upgrade the stack version to make the CI more reliable. @lalit-satapathy how about you?

@shmsr, Is the new conclusion that the CI is passing in 8.9.* and failing in 8.8.* for _dimension_fields_ override? Our earlier understanding as per [this](https://github.com/elastic/elastic-package/issues/1174#issuecomment-1637707840), lead to conclude that, the issue is resolved in 8.8.* onwards. 

I am fine, in that is case to upgrade to 8.9. Need to summarise and to see if any changes needed in other packages. Example [kafka](https://github.com/elastic/integrations/blob/main/packages/kafka/data_stream/consumergroup/manifest.yml#L30)?

CC: @agithomas

**Comment by shmsr** [comment]
> > i'm happy to upgrade the stack version to make the CI more reliable. @lalit-satapathy how about you?
> 
> @shmsr, Is the new conclusion that the CI is passing in 8.9.* and failing in 8.8.* for _dimension_fields_ override? Our earlier understanding as per [this](https://github.com/elastic/elastic-package/issues/1174#issuecomment-1637707840), lead to conclude that, the issue is resolved in 8.8.* onwards.
> 
> I am fine, in that is case to upgrade to 8.9. Need to summarise and to see if any changes needed in other packages. Example [kafka](https://github.com/elastic/integrations/blob/main/packages/kafka/data_stream/consumergroup/manifest.yml#L30)?
> 
> CC: @agithomas

For some unknown reason it is only happening for `influxdb` and this is the reason why I [pulled out](https://github.com/elastic/integrations/pull/8171#issuecomment-1768219287) `influxdb` from https://github.com/elastic/integrations/pull/8171 to unblock that PR from merging.

After failing to conclude why the CI failed for `influxdb`, I tried reproducing this locally, and there too the tests were failing with with 8.8.0 stack. But for 8.9.0 it passed.

**Comment by lalit-satapathy** [comment]
> For some unknown reason it is only happening for `influxdb` and this is the reason

So long as it consistently passes in 8.9 and fails in 8.8, we can conclude that some external change done in 8.9 stack.

**Pull request merged**